### ISSUE TITLE
Improve error stacktrace in the isolate implementation

### ISSUE THREAD

**Issue by davidmartos96** [simolus3/drift#1750]
Hello! While working with drift behind an isolate, it would be great if the stack trace from query errors would show the main isolate stack trace instead of the one in the background isolate. That way it would be easier to tell where the error occurred.

I think this could be possible with the new Dart 2.16 feature `Error.throwWithStackTrace(e, current)`, already in Flutter stable.

Right now, the stack traces are not really helpful:

```
[Flutter] │ #0   throwException (package:sqlite3/src/impl/exception.dart:34:3) package:sqlite3/…/impl/exception.dart:34
[Flutter] │ #1   DatabaseImpl._prepareInternal (package:sqlite3/src/impl/database.dart:250:9) package:sqlite3/…/impl/database.dart:250
[Flutter] │ #2   DatabaseImpl.prepare (package:sqlite3/src/impl/database.dart:152:19) package:sqlite3/…/impl/database.dart:152
[Flutter] │ #3   _VmDelegate.runSelect (package:drift/src/ffi/database.dart:231:22) package:drift/…/ffi/database.dart:231
[Flutter] │ #4   _BaseExecutor.runSelect.<anonymous closure> (package:drift/src/runtime/executor/helpers/engines.dart:71:19) package:drift/…/helpers/engines.dart:71
[Flutter] │ #5   _BaseExecutor._synchronized.<anonymous closure> (package:drift/src/runtime/executor/helpers/engines.dart:51:22) package:drift/…/helpers/engines.dart:51
[Flutter] │ #6   new Future.sync (dart:async/future.dart:296:31)
[Flutter] │ #7   Lock.synchronized.callBlockAndComplete (package:drift/src/utils/synchronized.dart:17:21) package:drift/…/utils/synchronized.dart:17
[Flutter] │ #8   Lock.synchronized.<anonymous closure> (package:drift/src/utils/synchronized.dart:21:35) package:drift/…/utils/synchronized.dart:21
[Flutter] │ #9   _rootRunUnary (dart:async/zone.dart:1434:47)
[Flutter] │ #10   _CustomZone.runUnary (dart:async/zone.dart:1335:19)
[Flutter] │ #11   <asynchronous suspension>
[Flutter] │ #12   _BaseExecutor.runSelect (package:drift/src/runtime/executor/helpers/engines.dart:68:20) package:drift/…/helpers/engines.dart:68
[Flutter] │ #13   <asynchronous suspension>
[Flutter] │ #14   ServerImplementation._runQuery (package:drift/src/remote/server_impl.dart:136:29) package:drift/…/remote/server_impl.dart:136
[Flutter] │ #15   <asynchronous suspension>

```

**Comment by simolus3** [comment]
Very nice idea! Actually using `Error.throwWithStackTrace(e, current)` wouldn't quite be enough since the flow is:

1. Isolate A sends a request, creates a `Completer` to store the result.
2. Isolate B encounters an exception while trying to handle the request, tells A about the exception and its stack trace.
3. Isolate A then completes the completer with an error.

So if we do `StackTrace.current` in step 3, the trace would only contain some internal methods responsible for receiving messages from receive ports.
Instead, we now capture the current stack trace in step 1 and throw with that trace if an error happens remotely.

**Comment by davidmartos96** [comment]
@simolus3 Oh, I see. Thanks for the explanation!
I've taken a look at 7b5c33593748c60831f1f3101a8581ff13b7b50e , to check out the implementation. Should `final foregroundTrace = _requestTraces[requestId];` be replaced with `final foregroundTrace = _requestTraces.remove(requestId);` ?
Otherwise traces would start piling up :smile:

**Comment by simolus3** [comment]
Yeah I've also noticed the memory leak and fixed it in 510aff3a603e2504f34f409d4b89f72c9bf9fa23.

**Comment by davidmartos96** [comment]
@simolus3 As a comeback to this, what is your opinion on overriding the "toString()" implementation for the `DriftRemoteException` class and return the internal `remoteCause` object as a String? That way, the error could be printed instead of:  `Instance of 'DriftRemoteException'`
